### PR TITLE
[www] Update Community Page

### PR DESCRIPTION
**Pull request by raskchanky**
Right now, pulling community events in does not happen automatically.  In the short term, we've hard coded the only upcoming event.  Once the JSON API plugin is live on the Chef Events site, we'll pull events in via JSONP.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-api/doc/api.raml [hunk 1]
```diff
@@ -241,6 +241,23 @@ securitySchemes:
                                 "id": "73089155726360582",
                                 "flags": 0
                             }
+/chef-events-feed:
```
Comment (reset) [inline components/builder-api/doc/api.raml]: I'd suggest naming this endpoint something a bit more RESTful. A plural representation of each object returned might be named something like `/community-events`

**Comment by ryankeairns** [comment]
Thank you!

I suspect I need to do some stuff to get the API running locally... if so, any existing docs on that?

File: components/builder-api/doc/api.raml (hunk 1 above)
Comment (raskchanky) [inline components/builder-api/doc/api.raml]: Will do!

**Comment by raskchanky** [comment]
@ryankeairns https://github.com/habitat-sh/habitat/blob/master/BUILDER_DEV.md

Also feel free to ping me with any questions or problems.

**Comment by raskchanky** [comment]
It sounds like marketing is going to install the JSON API Wordpress plugin on the events site, which should be done today or tomorrow.  With that in place, we should be able to fetch the events directly from the middleman javascript file, and all the Rust code in this PR can be eliminated.

Don't merge this until that lands, and I can make the appropriate changes.

**Comment by ryankeairns** [comment]
Just a heads up, I'm currently making the design/style changes on this branch as well.

**Comment by ryankeairns** [comment]
@raskchanky I did a few CSS things that won't make my family, or @cnunciato proud, but the time and urgency I think this will suffice :)

I expect this to fail the tests for what appears to be API stuff (that we don't need for now since I hard-coded in the only event - ChefConf - for now)... if you could take a peek at that, it would be much appreciated.

Thank you!

File: www/source/community.html.slim [hunk 2]
```diff
@@ -7,50 +7,95 @@ section.community--hero
...
+        h2 Upcoming Events
+        .row
+          .community--events--card.columns.medium-6.medium-push-3
+            .community--events--card-heading.heading
+              ul.no-bullet
+                li Meetup
+            .body-content
+              h4 ChefConf 2017
+              p
+                | Join us at ChefConf to learn aobut all things Habitat. There will be several presentations
```
Comment (cnunciato) [inline www/source/community.html.slim]: `about`

File: www/source/community.html.slim [hunk 3]
```diff
@@ -7,50 +7,95 @@ section.community--hero
...
+                  during the conference in addition to ongoing discussions in the Habitat zone.
+              .article-detail--meta
+                ul.no-bullet
+                  li
+                    img[src="/images/icons/icon-calendar.svg"
+                      onerror="this.src='/images/icons/icon-calendar.png'" alt="calendar icon"]
```
Comment (cnunciato) [inline www/source/community.html.slim]: This is safe, but I don't think it's necessary anymore; support for SVGs as `img` references is [pretty solid](http://caniuse.com/#feat=svg-img).

**Comment by raskchanky** [comment]
I'm going to remove the Rust API I built to make this work, since I think JSONP is a better way forward.  It's not working right now, but since there's only 1 upcoming event and it's already hard-coded in, I think this is probably fine for the short term.  Once the JSON API plugin is working on the Chef events site, we can circle back around and finish this off.

File: www/source/javascripts/community.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+$(function() {
+  if ($(".community-events")) {
+    var url = DEPOT_API + "/community-events";
+
+    $.ajax(url, {
+      success: function(data, status, xhr) {
+        if (typeof data === "object" && typeof data.length === "number") {
```
Comment (cnunciato) [inline www/source/javascripts/community.js]: You might consider  `if (Array.isArray(data))` here, if you're concerned about the data not being an array. (Although, should it ever not be?)

**Comment by cnunciato** [comment]
One little typo, but otherwise looks good to me! Once the server-side stuff and the depot-URL configs are removed, this should be good to go.

File: www/source/javascripts/community.js (hunk 4 above)
Comment (raskchanky) [inline www/source/javascripts/community.js]: You're right, it's probably always going to be an array, but on the off chance something weird ends up in there, I'd rather it fail gracefully.  Thanks for the tip about `Array.isArray`.  That's totally what I wanted.  I'll update it.

**Comment by cnunciato** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/234336499).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/234336499).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**